flink-core/src/main/java/org/apache/flink/api/connector/source/SourceReader.java

```diff
@@ -58,7 +60,7 @@
 	 *
 	 * @return the state of the source.
 	 */
-	List<SplitT> snapshotState();
+	List<SplitT> snapshotState(long checkpointId);
```

Comment:
Would it make sense to retain the old method too and add a default implementation for the new method so that it doesn't
a. break existing users
b. if users didn't care about persisting any state related to the checkpoint, then they don't have to implement this method?

```
List<SplitT> snapshotState(long checkpointId);

default List<SplitT> snapshotState(long checkpointId) {
  return snapshotState();
}
```

This would also be consistent with the way the enumerators/readers implement the checkpointlistener API optionally to be notified on checkpoint related events IMO.

Reply:
Fair enough. I agree that it makes sense to move it to the final state in one shot.